File: task/task-api/routes/get-employee.go
package routes

import (
	"github.com/EspressoTrip-v2/concept-go-common/utils"
	"github.com/gorilla/mux"
	"net/http"
	taskPackage "task-api/proto"
	"task-api/services/grpc"
	"task-api/tacer"
)

func GetEmployee(w http.ResponseWriter, r *http.Request) {
	ctx, span := tacer.GetTracer().Start(r.Context(), "GetEmployee")
	defer span.End()

	employeeId := mux.Vars(r)["employeeId"]
	employeeRequest := taskPackage.EmployeeRequest{
		Id: employeeId,
	}
	response, err := grpc.GrpcClient().GetEmployee(ctx, &employeeRequest)
	if err != nil {
		utils.WriteResponse(w, err.Status, err.GetErrors())
	} else {
		utils.WriteResponse(w, http.StatusOK, response)
	}
}

File: employee-dash/employee-dash-api/src/routes/get-employee.ts
import express, { Request, Response } from "express";
import { requireAuth } from "@espressotrip-org/concept-common";
import { GrpcEmployeeDashClient } from "../services";

const router = express.Router();

router.get("/api/employee/:employeeId", requireAuth, async (req: Request, res: Response) => {
    const id = <string>req.params.employeeId;
    const grpcResponse = await GrpcEmployeeDashClient.getClient().getEmployee(id);
    res.send(grpcResponse);
});

export { router as getEmployeeRouter };
